spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/PropertySourceUtils.java

```diff
@@ -54,20 +59,15 @@
 			return properties;
 		}
 		catch (IOException e) {
-			throw new IllegalArgumentException();
+			throw new UncheckedIOException(e);
 		}
 	};
 
 	/**
 	 * Function to convert Properties to a Map.
 	 */
 	public static final Function<Properties, Map<String, Object>> PROPERTIES_TO_MAP = p -> p.entrySet().stream()
-			.collect(Collectors.toMap(e -> String.valueOf(e.getKey()), Map.Entry::getValue, throwingMerger(),
```

Comment:
- I really dislike usages of `String::valueOf` for these cases, for the simple fact that internally (it is even documented) it will do a mapping of a `null` to a `"null"`. I would much rather prefer the code to fail at this point with a NPE, instead of this transformation. Instead I propose to use `e.getKey().toString()`
- then providing the `throwingMerger()` is not really needed, but, without such a merger, you can not specify a custom `Supplier` like `LinkedHashMap::new`. I perfectly understand that. If you take a step back, and think that the source of the stream is a `Properties` (a `Map`), that has _undefined order_, collecting this stream to a `LinkedHashMap` (that defines an order), makes no sense. As such, remove both the merger and the Supplier.